core/blockchain/impl/block_tree_impl.hpp

```diff
@@ -148,6 +149,16 @@ namespace kagome::blockchain {
     void removeUnfinalized() override;
 
    private:
+    struct BlocksPruning {
+      BlocksPruning(std::optional<uint32_t> config,
+                    primitives::BlockNumber finalized);
+
+      primitives::BlockNumber limitFor(primitives::BlockNumber finalized) const;
```

Comment:
Also requires documentation as naming is not straightforward.
Or call this method something like `getLowestBlockToPrune`